tortoise/rerun.go

```diff
@@ -31,7 +31,13 @@ func (t *ThreadSafeVerifyingTortoise) updateFromRerun(ctx context.Context) (bool
 			updated   = completed.Consensus
 			err       error
 		)
-		if current.Last.After(updated.Last) {
+		// we need to persist after rerun is completed so that eviction will not miss any layers
+		// that were received concurrently.
+		err = completed.Consensus.Persist()
```

Comment:
:s/completed.Consensus/updated/

Reply:
why `Persist` here instead of after `catchupToCurrent` is completed?

Reply:
because we are evicting layers based on the window, if state after rerun is not persisted we will evict some layers only from memory while doing a catchup and handling new layer